finagle-netty4-http/src/main/scala/com/twitter/finagle/netty4/http/handler/FixedLengthMessageAggregator.scala

```diff
@@ -5,10 +5,25 @@ import com.twitter.util.StorageUnit
 import io.netty.handler.codec.http._
 
 /**
- * netty can chunk fixed length messages so we dechunk them if the content-length
- * is sufficiently small so that streaming clients and servers can access message
- * contents via `content` and `contentString`. Chunked transfer encoded messages
- * still require accessing `reader`.
+ * Aggregates fixed length http messages and emits [[FullHttpMessage]] downstream.
+ *
+ * Http message is considered fixed length if it has `Content-Length` header and
+ * `Transfer-Encoding` header is not `chunked` or if we can deduce actual content
+ * length to be 0 even if `Content-Length` header is not specified (f.ex. from RFCs).
+ *
+ * Fixed length messages may arrive as a series of smaller pieces (hereinafter referred
+ * to as chunks, but not to be confused with http chunks) from the upstream handlers
+ * in pipeline. To eventually build a [[com.twitter.finagle.http.Request]] object with
+ * content available as [[com.twitter.finagle.http.Request.content]] or
+ * [[com.twitter.finagle.http.Request.contentString]] all chunks must be stored in
+ * temporary buffer and combined together into a [[FullHttpMessage]] as soon as the last
+ * chunk of the message is received.
+ *
+ * `maxContentLength` parameter sets the limit on message's `Content-Length`.
```

Comment:
I think this could be improved. Perhaps:

"The `maxContentLength` determines when to aggregate chunks and when to bypass the message as is. Only sufficiently small messages (smaller than `maxContentLength`) are aggregated."